src/thread.rs

```diff
@@ -711,7 +711,15 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn run_timeout_callback(&mut self) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
         let (thread, callback) =
-            this.machine.threads.get_ready_callback().expect("no callback found");
+            if let Some((thread, callback)) = this.machine.threads.get_ready_callback() {
+                (thread, callback)
+            } else {
+                // get_ready_callback can return None if the computer's clock was
+                // shifted after calling the scheduler and before the call
+                // to get_ready_callback. In this case, just do nothing, which
+                // effectively just returns to the scheduler.
```

Comment:
Could you add a link to https://github.com/rust-lang/miri/issues/1763?

Reply:
Done.